src/components/Toaster/Toast/Toast.tsx

```diff
@@ -37,39 +37,42 @@ enum ToastStatus {
 }
 
 interface UseToastHeightProps {
-    isOverride: boolean;
     status: ToastStatus;
+    updatesCounter: number;
 }
 
 function getHeight(ref: React.RefObject<HTMLDivElement>) {
     return ref.current?.offsetHeight;
 }
 
-function useToastHeight({isOverride, status}: UseToastHeightProps) {
+function useToastHeight({status, updatesCounter}: UseToastHeightProps) {
     const [height, setHeight] = React.useState<number | undefined>(undefined);
 
     const ref = React.useRef<HTMLDivElement>(null);
 
-    const heightRef = React.useRef<number>();
     React.useEffect(() => {
         // ATTENTION: getting `offsetHeight` is important for correct transaction of `height`
         // HOW THIS WORKS:
         // On the step of changing state to `ShowingIndent` we set class with height `transition`
         // We need now to apply this styles, to achieve this we're calling `offsetHeight`, to force repaint
         // Now we call changing state to `ShowingHeight` and changing height now happen with transition
         if (status === ToastStatus.ShowingIndents) {
-            heightRef.current = getHeight(ref);
+            getHeight(ref);
         }
     }, [status]);
 
     React.useEffect(() => {
-        const height =
-            typeof heightRef.current === 'number' && !isOverride
-                ? heightRef.current
-                : getHeight(ref);
+        if (
+            status === ToastStatus.ShowingIndents ||
+            status === ToastStatus.ShowingHeight ||
+            status === ToastStatus.Hiding
+        ) {
+            return; // Not update height during animation
+        }
 
-        setHeight(height);
-    }, [isOverride]);
+        const newHeight = getHeight(ref);
+        setHeight(newHeight);
+    }, [updatesCounter]);
```

Comment:
You want to update height on sequential updates?

Reply:
Yes. This calculated height is applied on the hiding stage.
If we set this height only on first update, there can be height jump on hiding animation.